core-bundle/src/Command/ResizeImagesCommand.php

```diff
@@ -0,0 +1,243 @@
+
+                return 0;
+            }
+
+            usleep((int) ($sleep * 1000000));
+
+            $maxDuration = max($maxDuration, $lastDuration = $this->resizeImage($path, $output));
+        }
+
+        $output->writeln("\n".'All images resized.');
+
+        return 0;
+    }
+
+    private function resizeImage(string $path, OutputInterface $output): float
+    {
+        $startTime = microtime(true);
+
+        if ($this->filesystem->exists($path)) {
+            return 0;
+        }
+
+        $output->write(str_pad($path, $this->terminalWidth - 14, '.').' ');
+
+        try {
+            $image = $this->imageFactory->create($this->targetDir.'/'.$path);
+            $resizer = $this->resizer;
+
+            if ($image instanceof DeferredImageInterface) {
+                $resizedImage = $resizer->resizeDeferredImage($image, false);
+
+                if (null === $resizedImage) {
+                    $output->write("\r".str_repeat(' ', $this->terminalWidth)."\r");
+                } else {
+                    $output->writeln(sprintf('done%7.3Fs', $duration = microtime(true) - $startTime));
+
+                    return $duration;
+                }
+            } else {
+                $output->write("\r".str_repeat(' ', $this->terminalWidth)."\r");
+            }
+        } catch (\Throwable $exception) {
+            $output->writeln('failed');
+            $output->writeln($exception->getMessage());
+        }
+
+        return 0;
+    }
+
+    private function executeConcurrent(int $count, InputInterface $input, OutputInterface $output): int
+    {
+        $phpFinder = new PhpExecutableFinder();
+
+        $output->writeln('Starting '.$count.' concurrent processes...');
+
+        if (false === ($phpPath = $phpFinder->find())) {
+            throw new \RuntimeException('The php executable could not be found.');
+        }
+
+        /** @var Process[] $processes */
+        $processes = [];
+
+        /** @var string[] $outputs */
+        $outputs = [];
+
+        for ($i = 0; $i < $count; ++$i) {
+            $process = new Process(array_merge(
+                [$phpPath],
+                $_SERVER['argv'],
+                ['--concurrent=1', $output->isDecorated() ? '--ansi' : '--no-ansi']
+            ));
+            $process->setTimeout(null);
+            $process->setEnv([
+                'LINES' => getenv('LINES'),
+                'COLUMNS' => getenv('COLUMNS') - 4,
+            ]);
+            $process->start();
+
+            $processes[] = $process;
+            $outputs[] = '';
+        }
+
+        do {
+            $isRunning = false;
+
+            foreach ($processes as $index => $process) {
+                if ($process->isRunning()) {
+                    $isRunning = true;
+                }
+                $outputs[$index] .= $process->getIncrementalOutput();
+                $rows = explode("\n", $outputs[$index]);
+                $outputs[$index] = array_pop($rows);
```

Comment:
`$rows` is already the variable for the temporary assignment. `$outputs[$index]` has to be reset in this line to make the code work. Would it help if I add a comment like `// Extract completed rows from the incremental output` above line 223?

Reply:
Can you explain what the code does? If I understand correctly, you're fetching the last row of the previous or current (incremental) output to show that?

Reply:
I fetch all rows except the last one, because the last row is possible incomplete an we should wait for further output.

Reply:
I see. Well that's really confusing (to me). Maybe you can use `$buffer` instead and add a comment what each line does. Example:

```
// Append new output to remaining buffer from previous loop run
$buffer[$index] .= $process->getIncrementalOutput();

// Split output rows
$rows = explode("\n", $buffer[$index]);

// Buffer and remove last line of output, as it might be incomplete
$buffer[$index] = array_pop($rows);

// Write remaining rows to the output with padded prefix for each thread
$output->write(/* ... */);
```

Reply:
4087595fbd9020486b882e922c1f7a0e8f0f033e